### ISSUE TITLE
Very slow performance on Android BiometricPrompt

### ISSUE THREAD

**Issue by vascofg** [oblador/react-native-keychain#309]
There's what I think is a bug on waiting for the Biometric Prompt result, more specifically in this part of the code:
https://github.com/oblador/react-native-keychain/blob/c854720a8945990dc75bbfbd02dc2044bda8aead/android/src/main/java/com/oblador/keychain/KeychainModule.java#L849-L859

Particularly in the case where there's an error in the flow, it takes several seconds for the thread to unblock.

I'll do some more investigation, but meanwhile you can try to reproduce:
- Trigger biometric prompt (e.g. via getInternetCredentials)
- Scan a wrong fingerprint a bunch of times, until you get a lockout
- Try to trigger biometric prompt again
  - It now takes several seconds to get an exception from the Promise

**Comment by enahum** [comment]
Yes, I'm also experiencing about 2 seconds performance degradation when getting credentials stored in Android, I even disabled the need for Biometrics to see if that made any difference and the reality is that it made no difference.

**Comment by lloisp** [comment]
Our Application experienced an `ApplicationNotResponding` (`android.os.BinderProxy` in `transactNative`), quoted on error-tracking as:
> Application Not Responding for at least 4000 ms.

So this seems to be more than just a simple delay and more like something that can hinder the user's experience.

**Comment by enahum** [comment]
@oblador thoughts about this?

**Comment by SMJ93** [comment]
We are experiencing this on some older Android devices:

Device: P2 Lite
Manufacturer: Shenzhen JEKO Communication Co.,Ltd.
OS: Android 7

```
Fatal Exception: java.util.concurrent.TimeoutException
android.security.keystore.AndroidKeyStoreUnauthenticatedAESCipherSpi$CBC$PKCS7Padding.finalize() timed out after 10 seconds
```

**Comment by emin93** [comment]
I have the same issue using the latest version (`7.0.0`) and can reproduce it sometimes when calling `getInternetCredentials` directly after the app start. "sometimes" meaning I restart the app over and over again and on every 5-7th app start it takes about 3 seconds until the prompt shows up.

**Comment by oblador** [comment]
Please try out 8.0.0 which has performance improvements

**Comment by emin93** [comment]
Thank you for the update. Unfortunately I can still reproduce the issue by restarting the app a few times. Every 4-6th app start it takes much longer until the prompt shows up (it's about the same as before with `7.0.0`).

Tried it now also on a Huawei P30, where it also happens.

To reproduce the issue, I simply have to call `getInternetCredentials` in an onMount `useEffect` hook. If it helps, I can create a reproduction repo and attach a video demonstrating the problem on a few devices. Please let me know if I should do so.

**Comment by DorianMazur** [comment]
This should be fixed with #658
